src/Extensions/Diagnostics.HealthChecks/HealthChecksBuilderExtensions.cs

```diff
@@ -34,13 +35,14 @@ public static class HealthChecksBuilderExtensions
 			string relativePath,
 			HttpMethod? method = null,
 			string? scheme = null,
+			IDictionary<string, string>? headers = null,
```

Comment:
https://docs.microsoft.com/en-us/dotnet/api/system.net.http.headers.httpheaders.add?view=net-5.0 Add accepts either a string or an IEnumerable.
Do we really want to restrict this here?